File: frontend/src/app/core/services/admin-background-jobs.service.ts
import { Injectable, inject } from '@angular/core';
import { ApiService } from './api.service';
import {
  Job,
  JobDetail,
  JobQueue,
  BackgroundJobsKpis,
} from '../../models/admin-background-jobs.model';

/**
 * Platform-staff API client for Background Jobs: the recurring-job registry (with run history),
 * KPIs, work-queue summaries, per-job detail, and run-now / pause / resume mutations. Reads need
 * INFRA_VIEW; mutations need INFRA_MANAGE (enforced + audit-logged on the backend).
 */
@Injectable({ providedIn: 'root' })
export class AdminBackgroundJobsService {
  private readonly api = inject(ApiService);
  private readonly base = '/admin/background-jobs';

  jobs() {
    return this.api.get<Job[]>(`${this.base}/jobs`);
  }

  kpis() {
    return this.api.get<BackgroundJobsKpis>(`${this.base}/kpis`);
  }

  queues() {
    return this.api.get<JobQueue[]>(`${this.base}/queues`);
  }

  getJob(id: string) {
    return this.api.get<JobDetail>(`${this.base}/jobs/${id}`);
  }

  runNow(id: string) {
    return this.api.post<Job>(`${this.base}/jobs/${id}/run`, {});
  }

  pause(id: string) {
    return this.api.post<Job>(`${this.base}/jobs/${id}/pause`, {});
  }

  resume(id: string) {
    return this.api.post<Job>(`${this.base}/jobs/${id}/resume`, {});
  }
}

File: backend/src/main/java/com/postwerk/service/AdminBackgroundJobsService.java
package com.postwerk.service;

import com.postwerk.dto.admin.BackgroundJobsKpisResponse;
import com.postwerk.dto.admin.JobDetailResponse;
import com.postwerk.dto.admin.JobQueueResponse;
import com.postwerk.dto.admin.JobResponse;

import java.util.List;
import java.util.UUID;

/**
 * Platform-staff Background Jobs: the platform's recurring scheduled jobs (real schedules + run
 * history from {@code job_runs}) and the work queues they drain, with gated run-now / pause / resume.
 * Reads need {@code INFRA_VIEW}; mutations need {@code INFRA_MANAGE} (enforced at the controller).
 *
 * @since 1.0
 */
public interface AdminBackgroundJobsService {

    List<JobResponse> listJobs();

    BackgroundJobsKpisResponse kpis();

    List<JobQueueResponse> queues();

    JobDetailResponse getJob(String id);

    /** Triggers a manual run of the job in the background; returns its current row. */
    JobResponse runNow(String id, UUID actorUserId, String ip);

    JobResponse setPaused(String id, boolean paused, UUID actorUserId, String ip);
}
